### PR TITLE
[BUG] Barter was allowing the purchase of bags. Bags could contain an…

### PR DESCRIPTION
**Pull request by fryguy503**
…ything and it would be traded.

Longterm to match Live's functionality, the search function in the buyer window should not return bags or no drop items at all.

### CODE REVIEW

File: zone/trading.cpp [hunk 1]
```diff
@@ -2602,6 +2602,11 @@ void Client::SellToBuyer(const EQApplicationPacket *app) {
 		return;
 	}
 
+    if(item->IsClassBag()) {
```
Comment (Akkadius) [inline zone/trading.cpp]: `if (` vs `if(`

File: zone/trading.cpp [hunk 2]
```diff
@@ -2602,6 +2602,11 @@ void Client::SellToBuyer(const EQApplicationPacket *app) {
...
+        Message(13, "That item is a Bag.");
```
Comment (Akkadius) [inline zone/trading.cpp]: Use `Chat::Red` vs `13`

File: zone/trading.cpp [hunk 3]
```diff
@@ -2973,6 +2978,9 @@ void Client::UpdateBuyLine(const EQApplicationPacket *app) {
 		else if(item->NoDrop == 0)
 			Message(Chat::Red, "Buy line %s disabled as the item is NODROP.", ItemName);
 
+        else if(item->IsClassBag())
+            Message(13, "Buy line %s disabled as the item is a Bag.", ItemName);
```
Comment (Akkadius) [inline zone/trading.cpp]: Use `Chat::Red` vs `13`

Comment (fryguy503) [reply zone/trading.cpp]: Got me on those, copied from my code pre Chat::Color

**Comment by Akkadius** [comment]
Still some weird formatting on this one; tagging @Uleat and @mackal for reviews

**Comment by mackal** [comment]
This is fine. It fixes the exploit, but as the PR says, it's not a complete fix. (This code should probably still exist just in case of packet forging non-sense I guess)

**Comment by Akkadius** [comment]
Seems like formatting got way weirder on this PR

**Comment by fryguy503** [comment]
> Seems like formatting got way weirder on this PR

Per Mackal I just reformatted the two functions and let CLion autoformat it all.

**Pull request closed without merging**